### PR TITLE
clean(autogpt): Remove old plugin system

### PR DESCRIPTION
**Pull request by kcze**
### Background

Follow up after merging https://github.com/Significant-Gravitas/AutoGPT/pull/7054, old plugins will no longer be used.

### Changes 🏗️

- Removed all dead code needed to load and use plugins.
- Removed `auto-gpt-plugin-template` dependency
- Removed `rev=` from `autogpt-forge` dependency (the set `rev` had incompatible `duckduckgo-search` versions)
- Kept `--install-plugin-deps` CLI option and dead code associated (may be needed for new plugins)

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

**Comment by kcze** [comment]
Do we want to keep `--install-plugin-deps` CLI option, assuming we want automatic deps install for new plugins? Can we make it a bool in config/env?

**Comment by Swiftyos** [comment]
@CodiumAI-Agent /review

File: autogpts/autogpt/.env.template [hunk 1]
```diff
@@ -44,9 +44,6 @@ OPENAI_API_KEY=your-openai-api-key
 ## AI_SETTINGS_FILE - Specifies which AI Settings file to use, relative to the AutoGPT root directory. (defaults to ai_settings.yaml)
 # AI_SETTINGS_FILE=ai_settings.yaml
 
-## PLUGINS_CONFIG_FILE - The path to the plugins_config.yaml file, relative to the AutoGPT root directory. (Default plugins_config.yaml)
```
Comment (ntindle) [inline autogpts/autogpt/.env.template:47]: Check docs for this too

**Review by ntindle — APPROVED** [review]

File: autogpts/autogpt/.env.template (hunk 1 above)
Comment (kcze) [reply autogpts/autogpt/.env.template:47]: done

**Pull request merged**